### PR TITLE
ISPN-7512 Add FRAG3 back to TCP configuration

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-7512

### CODE REVIEW

**Comment by alanfx** [comment]
👍 💯 🥇 from me!

**Comment by wburns** [comment]
👍 Seems good to me, but I would wait to see what @danberindei says.  I don't understand how it would work with a bigger message (larger than credit size) without FRAG.

**Comment by slaskawi** [comment]
@pruivo Just a silly question, shouldn't we run perf tests here?

**Comment by danberindei** [comment]
@slaskawi When the difference is between working and not-working, I'm not sure perf tests are relevant... either way, our perf tests use 10k values, so they would never be fragmented (the default `frag_size` is 60k).

I'd make `frag_size` even bigger, say 200k, to make fragmentation even less likely. For reference, `max_credits * min_threshold` is 800k in our default configs.

**Comment by pruivo** [comment]
updated the ```frag_size``` except for server configuration that uses ```jgroups-defaults.xml``` values.

**Comment by tristantarrant** [comment]
Aha, server does not have a way to distinguish between defaults for TCP and for UDP.

File: server/integration/jgroups/src/main/resources/subsystem-templates/infinispan-jgroups.xml [hunk 1]
```diff
@@ -39,6 +39,7 @@
                 <protocol type="pbcast.STABLE" />
                 <protocol type="pbcast.GMS" />
                 <protocol type="MFC" />
+                <protocol type="FRAG3" />
```
Comment (tristantarrant) [inline server/integration/jgroups/src/main/resources/subsystem-templates/infinispan-jgroups.xml]: Add the explicit frag size for TCP to override the UDP default.

File: server/integration/jgroups/src/main/resources/subsystem-templates/infinispan-jgroups.xml [hunk 2]
```diff
@@ -57,6 +58,7 @@
...
```
Comment (tristantarrant) [inline server/integration/jgroups/src/main/resources/subsystem-templates/infinispan-jgroups.xml]: Same here

**Comment by pruivo** [comment]
@tristantarrant added the explicit ```frag_size``` to that 2 places

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**